pkg/recommendation/replicas/prepare.go

```diff
@@ -0,0 +1,189 @@
+func (rr *ReplicasRecommender) CheckDataProviders(ctx *framework.RecommendationContext) error {
+	// 1. load data provider from recommendation config, override the default data source
+	configSet := rr.Recommender.Config
+	// replicas recommender only need history data provider
+	// History data source
+	// metricserver can't collect history data
+	// default is prometheus, you can override the provider to grpc or override the prometheus config
+	// TODO(xieydd) Load cache data source provider if config is not changed.
+	configKeys := config.GetKeysOfMap(configSet)
+	promKeys := fuzzy.FindFold(string(providers.PrometheusDataSource), configKeys)
+	dataSourceKeys := fuzzy.FindFold(providers.DataSourceTypeKey, configKeys)
+	//grpcKeys := fuzzy.FindFold(string(providers.GrpcDataSource), configKeys)
+
+	dataSourceType := dataSourceKeys[0]
+
+	if dataSourceType != string(providers.PrometheusDataSource) {
+		return fmt.Errorf("in replicas recommender, only suppport prometheus history data source")
+	}
+
+	if len(promKeys) != 0 {
+		return fmt.Errorf("in replicas recommender, you need set prometheus config %v for history data provider", providers.PrometheusConfigKeys)
+	}
+
+	mustSetConfig := []string{"prometheus-address", "prometheus-auth-username", "prometheus-auth-password", "prometheus-auth-bearertoken"}
+	if config.SlicesContainSlice(promKeys, mustSetConfig) {
+		return fmt.Errorf("in replicas recommender, you need set prometheus config %v for history data provider", mustSetConfig)
+	}
+	timeOut := 3 * time.Minute
+	if value, ok := configSet["prometheus-timeout"]; ok {
+		timeOut, _ = time.ParseDuration(value)
+	}
+
+	aliveTime := 60 * time.Second
+	if value, ok := configSet["prometheus-keepalive"]; ok {
+		aliveTime, _ = time.ParseDuration(value)
+	}
+
+	concurrency := 10
+	if value, ok := configSet["prometheus-query-concurrency"]; ok {
+		concurrency, _ = strconv.Atoi(value)
+	}
+
+	maxPoints := 11000
+	if value, ok := configSet["prometheus-maxpoints"]; ok {
+		maxPoints, _ = strconv.Atoi(value)
+	}
+	promConfig := providers.PromConfig{
+		Address:            configSet["prometheus-address"],
+		Timeout:            timeOut,
+		KeepAlive:          aliveTime,
+		InsecureSkipVerify: configSet["prometheus-insecure-skip-verify"] == "true",
+		Auth: providers.ClientAuth{
+			Username:    configSet["prometheus-auth-username"],
+			Password:    configSet["prometheus-auth-password"],
+			BearerToken: configSet["prometheus-auth-bearertoken"],
+		},
+		QueryConcurrency:            concurrency,
+		BRateLimit:                  configSet["prometheus-bratelimit"] == "true",
+		MaxPointsLimitPerTimeSeries: maxPoints,
+	}
+	promDataProvider, err := prom.NewProvider(&promConfig)
+	if err != nil {
+		return err
+	}
+	ctx.DataProviders = map[providers.DataSourceType]providers.History{
+		providers.PrometheusDataSource: promDataProvider,
+	}
+
+	// if no data provider config set, use default history data provider
```

Comment:
Do we need to keep this block?

Reply:
we does not need this block, i will delete it.